ls1intum/Artemis
# `Development`: Use posts for plagiarism cases

### Checklist
#### General



#### Client

### Motivation and Context


#7437 made the client use /messages endpoints instead of /posts endpoints in every case. However, for plagiarism notifications, the /posts endpoints have to be used.

### Description

The client has been adapted and now uses /messages only for messaging/communication related use cases and not plagiarism cases.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 course where all 3 users are a member of

Need to deploy on legacy ts1 or ts0 (for mail support)

1. Generate a plagiarism case:
      1. Create a text exercise with the instructor with a due date in the future.
      2. Participate with **two** students in the exercise and submit the same text of at least 100 characters. You can use [loremipsum.de](https://loremipsum.de).
      3. Wait for the due date or set it to "now" with the instructor
      4. With the instructor, open the exercise under course-management (click the exercise title or id). Then follow the steps Click "Check plagiarism" -> Click "Detect plagiarism" -> Select the result on the left -> Click "Confirm"
2. Assess the plagiarism case:
      1. With the instructor, navigate to "Assessment", and click the "Plagiarism Cases" button. Select one of the students. On the next page click "Notify student" and send the post.
      2. With the student, navigate to the same URL, but replace "course-management" with "courses".
      3. Confirm that you can send a reply as the student
3. Check that the communication/messaging feature still works:
      1. Navigate to the communication tab and check that you can still reply to messages
      2. Navigate to the messages tab and check that you can still send messages and reply to them
      3. Navigate to a lecture and exercise details page and check that you can still send messages and reply to them

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






Same as develop

### Screenshots


No UI changes
